16.10.17 notes (#11080)
chef/chef: RELEASE_NOTES.md

- Alibaba Cloud support with node['alibaba'] showing metadata, `alibaba?` helper and node['cloud'] returning data now - https://github.com/chef/chef/pull/11004
- macOS builds now use openSSL 1.1.1

## What's New in 16.10.17

### Bugfixes

- Resolved installation failures on some Windows systems

> - Fixed the mount resource mount to work with network mounts which use the root level as the device. Thanks [@ramereth](https://github.com/ramereth)! 

Reviewer: ```suggestion
- Fixed the `mount_resource` for network mounts using the root level as the device. Thanks [@ramereth](https://github.com/ramereth)! 
```

> - Resolved a failure in the compliance phase when the profile name includes an @ symbol.

Reviewer: ```suggestion
- Resolved a compliance phase failure with profile names using the `@` symbol.
```

### Security

> OpenSSL has been upgraded to 1.0.2y to resolve the following CVEs:

Reviewer: ```suggestion
Upgraded OpenSSL to 1.0.2y, which resolves the following CVEs:
```
